### PR TITLE
[feat] Switching from bytecode to mnemonic (#145)

### PR DESCRIPTION
**Pull request by jonahcui**
Hi, all. This pr will add support for translate between bytecode and mnemomics.

### CODE REVIEW

**Comment by dorlevi** [comment]
thanks again for your contributions!

Just played around with this branch and I'm getting an [Error] Invalid EVM version requested when running/deploying a contract, will look take another look later to find the issue

**Comment by jonahcui** [comment]
ok, I'll fix it soon.

File: components/Editor/index.tsx [hunk 1]
```diff
@@ -294,6 +298,26 @@ const Editor = ({ readOnly = false }: Props) => {
...
+      value &&
+      lastCodeType === CodeType.Bytecode &&
+      value === CodeType.Mnemonic &&
+      instructions?.length > 0
+    ) {
+      const code = getBytecodeLinesFromInstructions(instructions)
+      setCode(code)
+    }
+
+    if (
```
Comment (assimovt) [inline components/Editor/index.tsx:311]: should this be else if?

File: components/Editor/index.tsx [hunk 2]
```diff
@@ -294,6 +298,26 @@ const Editor = ({ readOnly = false }: Props) => {
       setCode(examples[value as CodeType][0])
     }
 
+    if (
```
Comment (assimovt) [inline components/Editor/index.tsx:301]: should this be else if

File: components/Editor/index.tsx [hunk 3]
```diff
@@ -284,6 +287,7 @@ const Editor = ({ readOnly = false }: Props) => {
 
   const handleCodeTypeChange = (option: OnChangeValue<any, any>) => {
     const { value } = option
+    const lastCodeType = codeType
```
Comment (assimovt) [inline components/Editor/index.tsx:290]: do you really need to redefine codeType?

File: components/Editor/index.tsx (hunk 2 above)
Comment (jonahcui) [reply components/Editor/index.tsx]: I've fixed it.

File: components/Editor/index.tsx (hunk 1 above)
Comment (jonahcui) [reply components/Editor/index.tsx]: I've fixed it.

File: components/Editor/index.tsx (hunk 3 above)
Comment (jonahcui) [reply components/Editor/index.tsx]: I remove the ```codeType``` and the statement that checks whether ``` lastCodeType``` is equal to bytecode or mnemonic in the if statement.

**Comment by peroket** [comment]
Hello, thanks for your contribution :-)
I think it's good, but from a user point of view I think it might be confusing, because first if you don't know it exists there's nothing telling you it will do that, and second because there is nothing to say if it actually worked or not.

So to integrate properly this feature, I think idealy it would be only on explicit request from the user, with possibly an error message in the console if it's not possible for some reason (like unknwon opcode). For the UI integration, I'm thinking next to the current drop down menu to switch language, adding another drop down menu with a "Convert to: X". Currently the option would be mnemonic to opcodes or the reverse, but that way we can imagine adding more, like solidity to any other and yul to opcode or mnemonic as well.

I know it's a lot to ask, so no worries if you can't do it :-)

**Comment by dorlevi** [comment]
thanks for making the fixes @jonahcui. I agree with @peroket  that to make this integrated and complete giving some indication if the feature worked in the console would be good. however, I don't believe we need an explicit dropdown just for this feature, in fact, I wouldn't want one even if you worked on it, so @jonahcui , don't worry about that.

@peroket, this feature is not going to be wildly used (assumption) or the main use case, so taking space in the UI and creating some clutter/more confusion for the main usecases for a secondary one is not something I'd like to do , also evm.codes is a pro-user tool, it's fine if there are some features that are not as discoverable and users can just find them by playing around or reading release notes/seeing on twitter, so all and all I prefer not to make it's own dropdown or anything like that, I can think about solutions that might both achieve discoverability while not adding more UI elements that won't be used often, such as a short console message "you can now switch between XXX and aYYY" or as a small banner but I wouldn't even bother on these. 

So to summarize, let's properly integrate this into the console and that's it.

P.S I am not too concerned that switching will not require an "explicit" action but will just work when switching the language selector, but part of how I would want to implement the console messages is that if it DOES fail, we will revert to showing the default mnemonic example, I would want to avoid situations where people switch to the mnemonic display and don't see anything because they had some malformed bytecode if that make sense.

**Comment by jonahcui** [comment]
@peroket @dorlevi OK, I see all the information. the key points are:
1. I agree that the function is not common, maybe it's just a small improvement
2. Add another dropdown is a good solution, but it can make users feel confused about why does there have two dropdowns. So I prefer the idea that adding a console message to adding another dropdown.

**Review by dorlevi — APPROVED** [review]

**Review by dorlevi — APPROVED** [review]
